packages/fuel-indexer-api-server/src/uses.rs

```diff
@@ -119,16 +120,38 @@ pub(crate) async fn health_check(
     })))
 }
 
-pub(crate) async fn stop_indexer(
+pub(crate) async fn remove_indexer(
     Path((namespace, identifier)): Path<(String, String)>,
     Extension(tx): Extension<Sender<ServiceRequest>>,
     Extension(pool): Extension<IndexerConnectionPool>,
     Extension(claims): Extension<Claims>,
+    params: Query<IndexerQueryParams>,
 ) -> ApiResult<axum::Json<Value>> {
     if claims.is_unauthenticated() {
         return Err(ApiError::Http(HttpError::Unauthorized));
     }
 
+    if params.stop_previous.unwrap_or(false) {
+        let mut conn = pool.acquire().await?;
+        let _ = queries::start_transaction(&mut conn).await?;
+
+        if let Err(e) =
+            queries::remove_previous_indexer(&mut conn, &namespace, &identifier).await
+        {
+            queries::revert_transaction(&mut conn).await?;
+
+            error!("Failed to remove Indexer({namespace}.{identifier}): {e}");
+
+            return Err(ApiError::Sqlx(sqlx::Error::RowNotFound));
```

Comment:
```suggestion
            return Err(ApiError::default());
```

Since we don't know that the error was `RowNotFound` for sure